autoload/crystal/indent.vim

```diff
@@ -70,29 +70,47 @@ let g:crystal#indent#crystal_deindent_keywords =
       \ g:crystal#indent#end_end_regex
 lockvar g:crystal#indent#crystal_deindent_keywords
 
+" Regex that defines a type declaration
+let g:crystal#indent#crystal_type_declaration =
+      \ '@\=\h\k*\s\+:\s\+\S.*'
+lockvar g:crystal#indent#crystal_type_declaration
+
 " Regex that defines continuation lines, not including (, {, or [.
-let g:crystal#indent#non_bracket_continuation_regex = '\%([\\.,:*/%+]\|\<and\|\<or\|\%(<%\)\@<![=-]\|\W[|&?]\|||\|&&\)\s*\%(#.*\)\=$'
+let g:crystal#indent#non_bracket_continuation_regex =
+      \ '\%(' .
+      \ '[\\.,:/%+\-=~<>|&]' .
+      \ '\|' .
+      \ '\W?' .
```

Comment:
Question: What does this `\W?` match?

Reply:
This is to match the `?` operator without matching the optional `?` in method names. For example, it should match `foo ?` for line continuations but not `foo?`.

Reply:
oh, ok. I understood. Thanks for the answer.